### PR TITLE
 Set up PostgreSQL DB for consumption

### PR DESCRIPTION
**Pull request by Gladdstone**
Spin up Docker containers containing base project structure. This PR also contains the initial DB scripts, but not the scripts to populate the database in the docker containers.

Issue  #2 

- [x] implement db in postgres
- [x] if actual db structure differs from documentation, update ERD
- [x] Test current structure
- [x] commit SQL files to repository

### CODE REVIEW

File: api/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM python:3.6
```
Comment (jrgabler) [inline api/Dockerfile:1]: Let's note this version in the
- Github Wiki under a file called setup 
-  our developement document

File: data/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
+FROM postgres:11
+LABEL maintainer "Joseph Farrell <joe@joefarrell.dev>"
+
+WORKDIR /data
+
+COPY create.sql /data
+COPY deployDb.sh /data
```
Comment (jrgabler) [inline data/Dockerfile:7]: Note: Will need to update once I commit a structure to the DB

File: data/create.sql [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        NEW.total_equity = NEW.quantity * NEW.purchase_price;
+        RETURN NEW;
+    END
+    $BODY$ LANGUAGE plpgsql;
+
+CREATE TRIGGER equity_trigger
+    BEFORE INSERT OR UPDATE
+    ON OwnedAssetsList
+    FOR EACH ROW
+        EXECUTE PROCEDURE computed_equity();
```
Comment (jrgabler) [inline data/create.sql:85]: Please update the ERD diagram in design document + add a footnote noting changes and date updated

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+version: '3.3'
```
Comment (jrgabler) [inline docker-compose.yml:1]: Like the Python version, please not Docker version in Github Wiki in setup page

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        build: ./api
+        ports:
+            - '5000:5000'
+
+    database:
+        image: postgres:11
+        restart: always
+        environment:
+            POSTGERS_USER: user
+            POSTGRES_PASSWORD: password
```
Comment (jrgabler) [inline docker-compose.yml:15]: so secure. much hacks. wow.

Comment (jrgabler) [reply docker-compose.yml:15]: Let's consider using `dotenv` or something similar and not commit our passwords to GitHub

I have a feeling that the professor _might_ be on our case about it. We can submit a dotenv via the mycourses dropbox :)

**Review by jrgabler — CHANGES_REQUESTED** [review]
Please also provide docker instructions for other team members (GitHub Wiki is great for that!)

Once documentation is up, I'll approve :)

File: docker-compose.yml (hunk 5 above)
Comment (Gladdstone) [reply docker-compose.yml:15]: Feature, not a bug, gonna refactor once we have functioning containers that you guys can build in

File: docker-compose.yml (hunk 4 above)
Comment (Gladdstone) [reply docker-compose.yml:1]: docker-compose, set

File: api/Dockerfile (hunk 1 above)
Comment (Gladdstone) [reply api/Dockerfile:1]: done

**Comment by Gladdstone** [comment]
Further DB population scripts coming in a later PR

**Review by jrgabler — APPROVED** [review]

**Pull request merged**